artifacts/ios_xcarchive_reader.go

```diff
@@ -1,14 +1,17 @@
-package zip
+package artifacts
 
-import "github.com/bitrise-io/go-xcode/plistutil"
+import (
+	"github.com/bitrise-io/go-xcode/plistutil"
+	"github.com/bitrise-io/go-xcode/v2/zip"
+)
 
 // IOSXCArchiveReader ...
 type IOSXCArchiveReader struct {
-	zipReader Reader
+	zipReader zip.ReadCloser
 }
 
 // NewIOSXCArchiveReader ...
-func NewIOSXCArchiveReader(reader Reader) IOSXCArchiveReader {
+func NewIOSXCArchiveReader(reader zip.ReadCloser) IOSXCArchiveReader {
```

Comment:
In this comment I am just thinking out loud. I looked at the other draft PR to see how this will be used and something caught my eye. 

At the moment the caller need to supply the zip reader. First it will try the default reader, then it needs to check if the error is a zip error and also check if ditto is available and then retry it with the ditto reader. 

My question is that does the caller need to know about all of these details? Or the `IOSXCArchiveReader` and `IPAReader` struct should handle these? 

At the moment I feel like that go-xcode package should take care of it. Like if ditto is available then always use that. If not then use the default one and also maybe print a warning message that it will fail with large zip archives. 

So to sum it up to me it looks like that this init method should be like a testing focused init method and maybe we should have another one which decides what zip struct to use so the caller should not care about it.

Reply:
Good suggestion, code updated in the following commits.